public/js/save.js

```diff
@@ -112,6 +154,10 @@ $(document).ready(function() {
     $("#clearForm").on('click', function() {
         $("#mainform")[0].reset();
     });
+    //Signs out of google
+    $("#signout").on('click', function() {
```

Comment:
Here or somewhere else, check if the user is logged in so you aren't making pointless, impossible requests. While that case /shouldn't/ ever arise, might as well handle it if it does.